src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -149,7 +149,7 @@ public function build(ContainerBuilder $container): void
             ]);
         }
 
-        if ($container->hasParameter('.kernel.config_dir') && $container->hasParameter('.kernel.bundles_definition')) {
+        if ('dev' === $container->getParameter('kernel.environment') && $container->getParameter('kernel.debug') && $container->hasParameter('.kernel.config_dir') && $container->hasParameter('.kernel.bundles_definition')) {
```

Comment:
we don't hardcode any env name in the codebase:
```suggestion
        if ($container->getParameter('kernel.debug') && $container->hasParameter('.kernel.config_dir') && $container->hasParameter('.kernel.bundles_definition')) {
```